extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/guice/MSQIndexingModule.java

```diff
@@ -188,14 +193,33 @@ public void configure(Binder binder)
 
   @Provides
   @LazySingleton
-  public Bouncer makeBouncer(final DruidProcessingConfig processingConfig, @Self final Set<NodeRole> nodeRoles)
+  public Bouncer makeBouncer(final DruidProcessingConfig processingConfig, Injector injector)
```

Comment:
In the current code, if the node roles are not set, does this `Set<NodeRole>` come out as empty or null?
Can't we just use a condition on that null or emptiness instead of using `Injector`?

Reply:
I think the issue is that there is nothing binded to `@self` annotation so the injector usage seems correct to me.

Reply:
For a generic fix, we could also try adding `ServerInjectorBuilder.registerNodeRoleModule` to the `HadoopTask` injector with empty role list. 
That will also require some change in `registerNodeRoleModule` to bind an empty set incase of an empty role list.

Reply:
the caller chain starts from a static initialization block in HadoopDruidIndexerConfig. It's hard to say where does this static block can be called from. (another reason to not like static initialization blocks)

Reply:
the `HadoopDruidIndexerConfig` injector would also bind an empty set of node roles if we fix `registerNodeRoleModule` to do so. The current code flow of `Initialization.makeInjectorWithModules` itself never passes any node roles to the injector

Reply:
Right. The current fix felt more generic to me. This is how it's done in the MetricsModule class as well. I am happy to do it that way too.